gutter/src/index.ts

```diff
@@ -10,18 +10,17 @@ import {EditorView} from "../../view/src"
 // FIXME at some point, add support for custom gutter space and
 // per-line markers
 
-// FIXME seriously slow on Firefox, quite fast on Chrome
+// FIXME seriously slow on Firefox when devtools are open
 
 export interface GutterConfig {
   fixed?: boolean,
   formatNumber?: (lineNo: number) => string
 }
 
-export function gutter(config: GutterConfig = {}) {
-  return new Plugin({
-    view(view: EditorView) { return new GutterView(view, config) }
-  })
-}
+export const gutter = Extension.defineUnique(configs => {
+  let config = combineConfig(configs)
+  return [viewPlugin.use(view => new GutterView(view, config))]
+}, {})
 
 class GutterView {
   dom: HTMLElement
```

Comment:
@marijnh I think it is better to extract the string as a field.
This is just my idea.

```javascript
private readonly GUTTER_ELEMENT_CLASS_NAME = "CodeMirror-gutter-element"

  dom: HTMLElement
  lineNo: number = -1
  height: number = -1
  above: number = -1
  below: number = -1

  constructor(lineNo: number, height: number, above: number, below: number, formatNo: (lineNo: number) => string) {
    this.dom = document.createElement("div")
    this.dom.className = this.GUTTER_ELEMENT_CLASS_NAME
    this.update(lineNo, height, above, below, formatNo)
  }
```

Reply:
@marijnh Am I commenting on a matter that is too small?

Reply:
Right, these kind of style nitpicks aren't all that helpful—I've usually put at least some thought into my default style. And regarding styling the editor, we're still [working out](https://github.com/codemirror/codemirror.next/issues/40) how to approach that, so these things are likely to be rewritten in the near future anyway.

Reply:
Thank you for your answer.